src/models/chainspec.rs

```diff
@@ -0,0 +1,394 @@
+    hardforks: HardForks,
+    params: Params,
+    genesis: Genesis,
+    precompiles: HashMap<String, HashMap<String, Precompiles>>,
+    balances: HashMap<H160, String>,
+}
+
+#[derive(Debug, Deserialize, PartialEq)]
+struct Engine {
+    name: String,
+    params: EngineParams,
+}
+
+#[derive(Debug, Deserialize, PartialEq)]
+struct EngineParams {
+    #[serde(deserialize_with = "deserialize_period_as_duration")]
+    period: Duration,
+    #[serde(deserialize_with = "deserialize_float_as_u64")]
+    epoch: u64,
+    genesis: EngineGenesis,
+}
+
+#[derive(Debug, Deserialize, PartialEq)]
+struct EngineGenesis {
+    vanity: H256,
+    signers: Vec<H160>,
+}
+
+#[derive(Debug, Deserialize, PartialEq)]
+struct HardForks {
+    eip140: U64,
+    eip145: U64,
+    eip150: U64,
+    eip155: U64,
+    eip160: U64,
+    eip161abc: U64,
+    eip161d: U64,
+    eip211: U64,
+    eip214: U64,
+    eip658: U64,
+    eip1014: U64,
+    eip1052: U64,
+    eip1283: U64,
+    eip1283_disable: U64,
+    eip1283_reenable: U64,
+    eip1344: U64,
+    eip1706: U64,
+    eip1884: U64,
+    eip2028: U64,
+    max_code_size: U64,
+}
+
+#[derive(Debug, Deserialize, PartialEq)]
+struct Params {
+    account_start_nonce: U64,
+    chain_id: U64,
+    gas_limit_bound_divisor: U64,
+    max_code_size: U64,
+    maximum_extra_data_size: U64,
+    min_gas_limit: U64,
+    network_id: U64,
+}
+
+#[derive(Debug, Deserialize, PartialEq)]
+struct Genesis {
+    gas_limit: U64,
+    timestamp: U64,
+}
+
+#[derive(Debug, Deserialize, PartialEq)]
+struct Precompiles {
+    name: String,
+    pricing: Pricing,
+}
+
+#[derive(Debug, Deserialize, PartialEq)]
+struct Pricing {
+    formula: String,
+    params: PricingParams,
+}
+
+#[derive(Debug, Deserialize, PartialEq)]
+#[serde(untagged)]
+enum PricingParams {
+    ModExp {
+        #[serde(deserialize_with = "deserialize_float_as_u64")]
+        divisor: u64,
+    },
+    Linear {
+        #[serde(deserialize_with = "deserialize_float_as_u64")]
+        base: u64,
+        #[serde(deserialize_with = "deserialize_float_as_u64")]
+        word: u64,
+    },
+    Price {
+        #[serde(deserialize_with = "deserialize_float_as_u64")]
+        price: u64,
+    },
+    BasePair {
+        #[serde(deserialize_with = "deserialize_float_as_u64")]
+        base: u64,
+        #[serde(deserialize_with = "deserialize_float_as_u64")]
+        pair: u64,
+    },
+    GasPerRound {
+        #[serde(deserialize_with = "deserialize_float_as_u64")]
+        gas_per_round: u64,
+    },
+}
+
+struct DeserializeF64asU64;
+
+impl<'de> de::Visitor<'de> for DeserializeF64asU64 {
+    type Value = u64;
+
+    fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+        formatter.write_str("an f64")
+    }
+
+    fn visit_f64<E>(self, v: f64) -> Result<Self::Value, E>
+    where
+        E: de::Error,
+    {
+        Ok(v as u64)
+    }
+}
+
+fn deserialize_float_as_u64<'de, D>(deserializer: D) -> Result<u64, D::Error>
```

Comment:
Remove, also move to use integers in Rinkeby reference itself.

Reply:
Sorry, I don't really understand what you mean here by "remove". How will this conversion be handled?

Reply:
There should be no conversion at all, please edit Rinkeby spec to e.g. replace `3000.0` with `3000`. I don't think we have any values with non-zero decimal places anywhere in the specs.